perffile/format.go

```diff
@@ -1027,7 +1028,7 @@ type RecordSample struct {
 	StackUser        []byte // if SampleFormatStackUser
 	StackUserDynSize uint64 // if SampleFormatStackUser
 
-	Weight  uint64  // if SampleFormatWeight
+	Weight  Weight  // if SampleFormatWeight || SampleFormatWeightStruct
```

Comment:
I'm kind of shocked that they changed the perf_events C API in a non-compatible way. I'm not necessarily opposed to changing our API in a non-backwards-compatible way, but I'm not sure it's actually the best approach. What would you think of instead having both an unchanged `Weight` field and a new `Weights` field, where which is filled out depends on the flags, just like the other fields? That seems more consistent and maintains backwards compatibility.

Another possible approach would be to change the type of Weight to something that's a named type for uint64, but has methods to extract the sub-fields, but that seems kind of messy (and is maybe sensitive to endianness?)

Reply:
It is not technically incompatible. They replaced a u64 with a 64-bit union. Comments somewhere also claim that the upper 32-bit of weight were completely unused, hence why they can reclaim them.

I'll split into two separate fields as you suggest, that does seem a bit cleaner.